include/ddc/kernels/splines/splines_linear_problem_2x2_blocks.hpp

```diff
@@ -0,0 +1,318 @@
+ * with all blocks except top-left one being stored in dense format.
+ *
+ * A = |   Q    | gamma |
+ *     | lambda | delta |
+ *
+ * The storage format is dense row-major for top-left, top-right and bottom-left blocks, and determined by 
+ * its type for the top-left block.
+ *
+ * This class implements a Schur complement method to perform a block-LU factorization and solve,
+ * calling top-left block and bottom-right block setup_solver() and solve() methods for internal operations.
+ *
+ * @tparam ExecSpace The Kokkos::ExecutionSpace on which operations related to the matrix are supposed to be performed.
+ */
+template <class ExecSpace>
+class SplinesLinearProblem2x2Blocks : public SplinesLinearProblem<ExecSpace>
+{
+public:
+    using typename SplinesLinearProblem<ExecSpace>::MultiRHS;
+    using SplinesLinearProblem<ExecSpace>::size;
+
+protected:
+    std::shared_ptr<SplinesLinearProblem<ExecSpace>> m_top_left_block;
+    Kokkos::View<double**, Kokkos::HostSpace> m_top_right_block;
+    Kokkos::View<double**, Kokkos::HostSpace> m_bottom_left_block;
+    std::shared_ptr<SplinesLinearProblem<ExecSpace>> m_bottom_right_block;
+
+public:
+    /**
+     * @brief SplinesLinearProblem2x2Blocks constructor.
+     *
+     * @param mat_size The size of one of the dimensions of the square matrix.
+     * @param q A pointer toward the top-left SplinesLinearProblem.
+     */
+    explicit SplinesLinearProblem2x2Blocks(
+            std::size_t const mat_size,
+            std::unique_ptr<SplinesLinearProblem<ExecSpace>> q)
+        : SplinesLinearProblem<ExecSpace>(mat_size)
+        , m_top_left_block(std::move(q))
+        , m_top_right_block(
+                  "top_right_block",
+                  m_top_left_block->size(),
+                  mat_size - m_top_left_block->size())
+        , m_bottom_left_block(
+                  "bottom_left_block",
+                  mat_size - m_top_left_block->size(),
+                  m_top_left_block->size())
+        , m_bottom_right_block(
+                  new SplinesLinearProblemDense<ExecSpace>(mat_size - m_top_left_block->size()))
+    {
+        assert(m_top_left_block->size() <= mat_size);
+
+        Kokkos::deep_copy(m_top_right_block, 0.);
+        Kokkos::deep_copy(m_bottom_left_block, 0.);
+    }
+
+    virtual double get_element(std::size_t const i, std::size_t const j) const override
+    {
+        assert(i < size());
+        assert(j < size());
+
+        std::size_t const nq = m_top_left_block->size();
+        if (i < nq && j < nq) {
+            return m_top_left_block->get_element(i, j);
+        } else if (i >= nq && j >= nq) {
+            return m_bottom_right_block->get_element(i - nq, j - nq);
+        } else if (j >= nq) {
+            return m_top_right_block(i, j - nq);
+        } else {
+            return m_bottom_left_block(i - nq, j);
+        }
+    }
+
+    virtual void set_element(std::size_t const i, std::size_t const j, double const aij) override
```

Comment:
```suggestion
    void set_element(std::size_t const i, std::size_t const j, double const aij) override
```

Reply:
I do the change but I probably need to revert it in the MR on SplinesLinearProblemPeriodicBand